include/RDM/RDMBuilder.hpp

```diff
@@ -1,48 +1,53 @@
 #ifndef GQCG_RDMBUILDER_HPP
 #define GQCG_RDMBUILDER_HPP
 
+#include "RDM/BaseRDMBuilder.hpp"
+#include "FockSpace/FockSpace.hpp"
+#include "FockSpace/FockSpaceProduct.hpp"
 
-#include "RDM/OneRDM.hpp"
-#include "RDM/TwoRDM.hpp"
-#include "RDM/RDMs.hpp"
-#include "FockSpace/BaseFockSpace.hpp"
+#include <memory>
 
 
 namespace GQCG {
 
 
 /**
- *  RDMBuilder is an abstract base class for the calculation of a density matrix from a given wave function
- *  or coefficient expansion in a Fock space
+ *  RDMBuilder is a wrapper around the RDMBuilders that provides the functionality of the correct RDMBuilder
+ *  for a given Fock space at compile- or runtime.
  */
+
+
 class RDMBuilder {
+private:
+    std::shared_ptr<GQCG::BaseRDMBuilder> rdm_builder;
 public:
     // CONSTRUCTOR
-    RDMBuilder() = default;
+    /**
+     *  Allocates a DOCIRDMBuilder based on @param fock_space
```

Comment:
```suggestion
     *  Allocates a DOCIRDMBuilder based on the nature of the given @param fock_space
```

You could opt to delete the whitelines and documentation above the other constructors below.